src/libraries/Common/src/System/Security/Cryptography/ECDsaCng.SignVerify.cs

```diff
@@ -48,12 +46,57 @@ public override byte[] SignHash(byte[] hash)
             }
         }
 
+#if INTERNAL_ASYMMETRIC_IMPLEMENTATIONS
+        public override bool TrySignHash(ReadOnlySpan<byte> source, Span<byte> destination, out int bytesWritten)
+        {
+            return TrySignHashCore(
+                source,
+                destination,
+                DSASignatureFormat.IeeeP1363FixedFieldConcatenation,
+                out bytesWritten);
+        }
+
+        protected override unsafe bool TrySignHashCore(
+            ReadOnlySpan<byte> hash,
+            Span<byte> destination,
+            DSASignatureFormat signatureFormat,
+            out int bytesWritten)
+        {
+#else
         public override unsafe bool TrySignHash(ReadOnlySpan<byte> source, Span<byte> destination, out int bytesWritten)
         {
+            ReadOnlySpan<byte> hash = source;
+#endif
             using (SafeNCryptKeyHandle keyHandle = GetDuplicatedKeyHandle())
             {
-                return keyHandle.TrySignHash(source, destination, AsymmetricPaddingMode.None, null, out bytesWritten);
+                if (!keyHandle.TrySignHash(hash, destination, AsymmetricPaddingMode.None, null, out bytesWritten))
+                {
+                    bytesWritten = 0;
+                    return false;
+                }
+            }
+
+#if INTERNAL_ASYMMETRIC_IMPLEMENTATIONS
+            if (signatureFormat == DSASignatureFormat.IeeeP1363FixedFieldConcatenation)
+            {
+                return true;
+            }
+
+            if (signatureFormat != DSASignatureFormat.Rfc3279DerSequence)
+            {
+                Debug.Fail($"Missing internal implementation handler for signature format {signatureFormat}");
+                throw new CryptographicException(
+                    SR.Cryptography_UnknownSignatureFormat,
+                    signatureFormat.ToString());
             }
+
+            return AsymmetricAlgorithmHelpers.TryConvertIeee1363ToDer(
+                destination.Slice(0, bytesWritten),
+                destination,
```

Comment:
Was it intended that these buffers overlap? Is the underlying behavior well-defined for this scenario?

Reply:
The calculation uses an intermediate already, so it's safe to overlap (it's intended for this usage)